lib/tasks/update_supplier_for_moves.rake

```diff
@@ -0,0 +1,28 @@
+# frozen_string_literal: true
+
+namespace :moves do
+  desc 'Update the suppliers for all the moves based on the effective dates of the Suppliers'
+  task update_suppliers: :environment do
+    total = Move.count
+
+    i = 0
+
+    puts 'Updating suppliers for moves...'
+    Move.find_each(batch_size: 200) do |move|
+      move.determine_supplier # TODO: this might need to change
```

Comment:
This would have worked with the `force` parameter, but I removed that so instead this will need to replicate some code from that method. Something like:

```ruby
effective_date = move.date.presence || move.date_from
move.supplier = SupplierChooser.new(effective_date, move.from_location).call
```

Reply:
It would be really good to flag up any moves that we can't find a suitable supplier for - this would indicate some missing data in `SupplierLocation`, so perhaps log those to the console with the relevant move reference, effective date and from location? Then we can investigate any of those we see when running the rake task. The theory is that this should run and if everything works, every move will have a supplier id assigned, but if there is missing data we need to know.

Reply:
I agree! we should log the moves in which is not possible to get the supplier id, and figuring out why.